lib/driver.js

```diff
@@ -143,7 +143,12 @@ class AndroidDriver extends BaseDriver {
       await this.startAndroidSession(this.opts);
       return [sessionId, this.caps];
     } catch (e) {
-      await this.deleteSession();
+      // ignoring delete session exception if any and throw the real error
+      // that happened while creating the session.
+      try {
+        await this.deleteSession();
+      } catch (ignored) {
```

Comment:
for compactness I often write this as:

```js
} catch (ign) {}
```

But that's a minor stylistic issue